File: native_security/ledger_keyring/android/src/main/kotlin/com/chapter2/ledger/models/LedgerDeviceModel.kt
package com.chapter2.ledger.models

public enum class LedgerDeviceModel(
    val displayName: String,
    val supportsBle: Boolean,
    val supportsTouchScreen: Boolean,
    val screenWidthPixels: Int,
    val serviceUuidString: String
) {
    NANO_S(
        displayName = "Nano S",
        supportsBle = false,
        supportsTouchScreen = false,
        screenWidthPixels = 128,
        serviceUuidString = ""
    ),
    NANO_SP(
        displayName = "Nano S Plus",
        supportsBle = false,
        supportsTouchScreen = false,
        screenWidthPixels = 128,
        serviceUuidString = ""
    ),
    NANO_X(
        displayName = "Nano X",
        supportsBle = true,
        supportsTouchScreen = false,
        screenWidthPixels = 128,
        serviceUuidString = "13D63400-2C97-0004-0000-4C6564676572"
    ),
    STAX(
        displayName = "Ledger Stax",
        supportsBle = true,
        supportsTouchScreen = true,
        screenWidthPixels = 400,
        serviceUuidString = "13D63400-2C97-6004-0000-4C6564676572"
    ),
    FLEX(
        displayName = "Ledger Flex",
        supportsBle = true,
        supportsTouchScreen = true,
        screenWidthPixels = 480,
        serviceUuidString = "13D63400-2C97-6004-0000-4C6564676572"
    );

    companion object {
        fun fromDisplayName(name: String): LedgerDeviceModel? {
            return entries.find { it.displayName.equals(name, ignoreCase = true) }
        }
    }
}

File: packages/ledger_keyring/ios/Classes/Models/LedgerDeviceModel.swift
import Foundation

public enum LedgerDeviceModel: String, CaseIterable, Codable, Sendable {
    case nanoS = "Nano S"
    case nanoSP = "Nano S Plus"
    case nanoX = "Nano X"
    case stax = "Ledger Stax"
    case flex = "Ledger Flex"

    public var supportsBluetooth: Bool {
        switch self {
        case .nanoX, .stax, .flex:
            return true
        case .nanoS, .nanoSP:
            return false
        }
    }

    public var supportsTouchScreen: Bool {
        switch self {
        case .stax, .flex:
            return true
        case .nanoS, .nanoSP, .nanoX:
            return false
        }
    }

    public var screenWidthPixels: Int {
        switch self {
        case .nanoS:
            return 128
        case .nanoSP, .nanoX:
            return 128
        case .stax:
            return 400
        case .flex:
            return 480
        }
    }

    public var serviceUUIDString: String {
        switch self {
        case .nanoX:
            return "13D63400-2C97-0004-0000-4C6564676572"
        case .stax, .flex:
            return "13D63400-2C97-6004-0000-4C6564676572"
        case .nanoS, .nanoSP:
            return ""
        }
    }
}
